src/collision/Distance.ts

```diff
@@ -707,3 +716,164 @@ Distance.Input = DistanceInput;
 Distance.Output = DistanceOutput;
 Distance.Proxy = DistanceProxy;
 Distance.Cache = SimplexCache;
+
+/**
+ * Input parameters for ShapeCast
+ */
+export class ShapeCastInput {
+  proxyA: DistanceProxy = new DistanceProxy();
+  proxyB: DistanceProxy = new DistanceProxy();
+  transformA: Transform | null = null;
+  transformB: Transform | null = null;
+  translationB: Vec2 = Vec2.zero();
+}
+
+/**
+ * Output results for b2ShapeCast
+ */
+export class ShapeCastOutput {
+  point: Vec2 = Vec2.zero();
+  normal: Vec2 = Vec2.zero();
+  lambda: number;
+  iterations: number;
+}
+
+/**
+ * Perform a linear shape cast of shape B moving and shape A fixed. Determines
+ * the hit point, normal, and translation fraction.
+ */
+//
+// GJK-raycast
+// Algorithm by Gino van den Bergen.
+// "Smooth Mesh Contacts with GJK" in Game Physics Pearls. 2010
+export const ShapeCast = function(output: ShapeCastOutput, input: ShapeCastInput): boolean {
+  output.iterations = 0;
+  output.lambda = 1.0;
+  output.normal.setZero();
+  output.point.setZero();
+
+  const proxyA = input.proxyA;
+  const proxyB = input.proxyB;
+
+  const radiusA = Math.max(proxyA.m_radius, Settings.polygonRadius);
+  const radiusB = Math.max(proxyB.m_radius, Settings.polygonRadius);
+  const radius = radiusA + radiusB;
+
+  const xfA = input.transformA;
+  const xfB = input.transformB;
+
+  const r = input.translationB;
+  const n = Vec2.zero();
+  let lambda = 0.0;
+
+  // Initial simplex
+  const simplex = new Simplex();
+  simplex.m_count = 0;
+
+  // Get simplex vertices as an array.
+  const vertices = simplex.m_v;
+
+  // Get support point in -r direction
+  const indexA = proxyA.getSupport(Rot.mulTVec2(xfA.q, Vec2.neg(r)));
```

Comment:
BTW: In theory we could also write
```
const indexA = proxyA.getSupport(Rot.mulTVec2(xfA.q, r).neg());
```
Creates a Vec2 object less, but I don't think its worth it?

Reply:
Yes, it's ok, I'm changing vector functions anyway